src/KubernetesClient/Informers/ResourceObservableExtensions.cs

```diff
@@ -0,0 +1,294 @@
+                                        !addedKeys.Contains(keySelector(x)) &&
+                                        !deletedKeys.Contains(keySelector(x)) &&
+                                        !updatedKeys.Contains(keySelector(x)))
+                            .ToReset()
+                            .ToList();
+
+                        return deletedEntities
+                            .Union(addedEntities)
+                            .Union(updatedEntities)
+                            .Union(resetEntities);
+                    })
+                    .Do(_ => Console.WriteLine(_))
+                    .SynchronizeCache(cache, keySelector)
+                    .Do(msg => { cacheSynchronized = true; })
+                    .Select(x => x.Value)
+                    .ObserveOn(Scheduler.Immediate)
+                    .Subscribe(observer);
+            });
+        }
+
+        /// <summary>
+        ///     Injects a <see cref="ResourceEvent{T}" /> of type <see cref="ResourceStreamType.Sync" /> into the observable for each item produced
+        ///     by the <see cref="ResourceStreamType.List" /> operation from <paramref name="source" />
+        /// </summary>
+        /// <param name="source">The source sequence that will have sync messages appended</param>
+        /// <param name="timeSpan">The timespan interval at which the messages should be produced</param>
+        /// <typeparam name="T">The type of resource</typeparam>
+        /// <returns>Original sequence with resync applied</returns>
+        public static IObservable<ResourceEvent<T>> Resync<T>(this IObservable<ResourceEvent<T>> source, TimeSpan timeSpan)
+        {
+            return Observable.Create<ResourceEvent<T>>(observer =>
+            {
+                var timerSubscription = Observable
+                    .Interval(timeSpan)
+                    .SelectMany(_ => source
+                        .TakeUntil(x => x.EventFlags.HasFlag(EventTypeFlags.ResetEnd))
+                        .Do(x =>
+                        {
+                            if (!x.EventFlags.HasFlag(EventTypeFlags.Reset))
+                                throw new InvalidOperationException("Resync was applied to an observable sequence that does not issue a valid List event block when subscribed to");
+                        })
+                        .Select(x => x.Value.ToResourceEvent(EventTypeFlags.Sync)))
+                    .Subscribe(observer);
+                // this ensures that both timer and upstream subscription is closed when subscriber disconnects
+                var sourceSubscription = source.Subscribe(
+                    observer.OnNext,
+                    observer.OnError,
+                    () =>
+                    {
+                        observer.OnCompleted();
+                        timerSubscription.Dispose();
+                    });
+                return new CompositeDisposable(timerSubscription, sourceSubscription);
```

Comment:
Consider `StableCompositeDisposable.Create`.

Reply:
Changed, but can you please explain the difference between these two? I read the description of the class, but I still don't understand what the difference is from a functional perspective.

Reply:
`CompositeCollection` is intended for scenarios where the number of disposables is not fixed; they can be arbitrarily added, removed, cleared etc.

`StableCompositeDisposable` is intended for scenarios where the number is fixed. `Create` in this instance [returns](https://github.com/dotnet/reactive/blob/08b377703c1976601c63c93871450738c00898bb/Rx.NET/Source/src/System.Reactive/Disposables/StableCompositeDisposable.cs#L33) an implementation specialised for two disposables, thereby eliminating several array allocations when compared to `CompositeCollection`.